tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs

```diff
@@ -209,8 +214,8 @@ public void DefaultSetting(Func<TracerSettings, object> settingGetter, object ex
                 settings = GetTracerSettings(key, value);
             }
 
-            object actualValue = settingGetter(settings);
-            Assert.Equal(expectedValue, actualValue);
+            var actualValue = settingGetter(settings);
+            actualValue.Should().BeEquivalentTo(expectedValue);
```

Comment:
`Be()` results in tests failing with:
```
Expected actualValue to be {["k1"] = "v1", ["k2"] = "v2"}, but found {["k2"] = "v2", ["k1"] = "v1"}.
```
because the keys are out of order.

Reply:
From IntelliSense remarks:

> Objects are equivalent when both object graphs have equally named properties with the same value, irrespective of the type of those objects. Two properties are also equal if one type can be converted to another and the result is equal. The type of a collection property is ignored as long as the collection implements `IEnumerable<out T>` and all items in the collection are structurally equal. Notice that actual behavior is determined by the global defaults managed by `AssertionOptions`.